src/cephadm/cephadm

```diff
@@ -107,6 +107,201 @@ class Ceph(object):
+            logger.debug('firewalld.service is not enabled')
+            return
+        if state != "running":
+            logger.debug('firewalld.service is not running')
+            return
+
+        logger.info("firewalld ready")
+        self.available = True
+
+    def enable_service_for(self, daemon_type):
+        # type (str) -> None
+        if self.disabled:
+            return
+        if not self.available:
+            logger.debug('Not possible to enable service <%ss>. firewalld.service is not available' % daemon_type)
+            return
+
+        fw_services = []
+        if daemon_type == 'mon':
+            fw_services.append('ceph-mon')
+        elif daemon_type in ['mgr', 'mds', 'osd']:
+            fw_services.append('ceph')
+        elif daemon_type == NFSGanesha.daemon_type:
+            fw_services.append('nfs')
+
+        for svc in fw_services:
+            out, err, ret = call([self.cmd, '--permanent', '--query-service', svc], verbose_on_failure=False)
+            if ret:
+                logger.info('Enabling firewalld service %s in current zone...' % svc)
+                out, err, ret = call([self.cmd, '--permanent', '--add-service', svc])
+                if ret:
+                    raise RuntimeError(
+                        'unable to add service %s to current zone: %s' % (svc, err))
+            else:
+                logger.debug('firewalld service %s is enabled in current zone' % svc)
+
+    def open_ports(self, fw_ports):
+         # type (List[int]) -> None
+        if self.disabled:
+            return
+        if not self.available:
+            logger.debug('Not possible to open ports <%s>. firewalld.service is not available' % fw_ports)
+            return
+
+        for port in fw_ports:
+            tcp_port = str(port) + '/tcp'
+            out, err, ret = call([self.cmd, '--permanent', '--query-port', tcp_port], verbose_on_failure=False)
+            if ret:
+                logger.info('Enabling firewalld port %s in current zone...' % tcp_port)
+                out, err, ret = call([self.cmd, '--permanent', '--add-port', tcp_port])
+                if ret:
+                    raise RuntimeError('unable to add port %s to current zone: %s' %
+                                    (tcp_port, err))
+            else:
+                logger.debug('firewalld port %s is enabled in current zone' % tcp_port)
+
+    def apply_rules(self):
+         # type () -> None
+        if self.disabled or not self.available:
+            return
+
+        call_throws([self.cmd, '--reload'])
+
+class Cli(object):
+    def __init__(self, admin_key, uid, gid, config, log_dir, timeout=None):
+        # type: (str,int, int, str,str, Optional[int]) -> None
+        self.admin_key = admin_key
+        self.uid = uid
+        self.gid = gid
+        self.config = config
+        self.log_dir = log_dir
+        self.timeout = timeout
+
+    def ceph(self, cmd, extra_mounts={}, timeout=DEFAULT_TIMEOUT):
```

Comment:
ha! my renaming this to `__call__`, you can keep the usage compatible.

Reply:
Yes... but i like more to see things like:
`cli.ceph(['mgr', 'module', 'enable', 'dashboard'])`
instead of:
`cli(['mgr', 'module', 'enable', 'dashboard'])`

IMHO, It seems more intuitive, and opens the door to add new methods in the Cli class to execute other kind of tools not using the ceph container.